Level.py

```diff
@@ -1,15 +1,210 @@
+        """
+        get wins path and conditions
+        :type: void
+        """
+        for indx in range(self.ymax):
+            for indy in range(self.xmax):
+                if self.map_canvas.map_matrix[indy][indx]['type'] == 'A':
+                    self.win_path.append([indx, indy])
+
+    def test_win_conditions(self):
+        """
+        test if players win
+        :rtype: boolean
+        """
+        for entry in self.win_path:
+            if entry not in self.player_path:
+                return False
+        return True
+
+    """"
+    Touch methods
+    """""
+
+    def on_touch_down(self, touch):
+
+        # sauvegarde le touch
+        ud = touch.ud
+        ud['identifier'] = str(touch.uid)
+        self.point = [touch.x, touch.y]
+        self.player_path = []
+
+        # dessine le point si l'emplacement est valide sinon supprime le dessin.
+        tile_type = self.get_tile_properties()
+        if self.get_touch_auth(tile_type):
+            with self.canvas:
+                Color(self.color)
+                ud['points'] = [Point(points=self.point, pointsize=self.touch_width, group=ud['identifier'])]
+        else:
+            touch.ungrab(self)
+            self.canvas.remove_group(ud['identifier'])
+            return
+
+        # sauvegarde les differentes coordonnées
+        self.old_point = ud['points'][0].points[:]
+        self.point = ud['points'][0].points[:]
+
+        # lance le touch
+        touch.grab(self)
+
+    def on_touch_move(self, touch):
+
+        # si le touch est lancé
+        if touch.grab_current is not self:
+            return
+
+        # sauvegarde le touch
+        ud = touch.ud
+        ud['identifier'] = str(touch.uid)
+        self.point = [touch.x, touch.y]
+
+        """
+        enlever le commentaire pour reimplementer le touch guider.
+
+        # recupere la distance relative entre les deux états du touch
+        x_dif = touch.x - self.old_point[0]
+        y_dif = touch.y - self.old_point[1]
+
+        # trouve la direction du touch guidé et sauvegarde les valeurs
+        if abs(x_dif) > abs(y_dif):
+            self.point[0] += x_dif
+        else:
+            self.point[1] += y_dif
+        """
+
+        # dessine le point si l'emplacement est valide sinon supprime le dessin.
+        tile_type = self.get_tile_properties()
+        if self.get_touch_auth(tile_type):
+            with self.canvas:
+                Color(self.color)
+                ud['points'] = [Point(points=self.point, pointsize=self.touch_width, group=ud['identifier'])]
+        else:
+            touch.ungrab(self)
+            self.canvas.remove_group(ud['identifier'])
+            return
+
+        # recupere les coordonnées
+        self.old_point = [touch.x, touch.y]
+
+    def on_touch_up(self, touch):
+
+        # si le touch est lancé
+        if touch.grab_current is not self:
+            return
+
+        if self.test_win_conditions:
+            print("win")
```

Comment:
I change with a comment but we need to keep it because we wait for levels implementations and menu to know what to display